plone/namedfile/field.py

```diff
@@ -106,7 +106,12 @@ def __init__(self, **kw):
         super().__init__(schema=self.schema, **kw)
 
     def _validate(self, value):
-        super()._validate(value)
+        # we don't want default validation of super()._validate(value)
+        # as that will read the whole file into memory
+        # schema has to be provided by value
+        if not self.schema.providedBy(value):
+            raise SchemaNotProvided(self.schema, value).with_field_and_value(
+                self, value)
```

Comment:
@davisagli  This works and I've implemented a version below but it still doesn't feel right since its not validating a data attribute exists or getSize exists etc. Also I had to remove IFile from INamedBlobFile to make it work which means any code checking for that will fail.

Ultimately the real problem is using @property for data in the case of blobs is too clever and should have a different interface that returns a file like object. Or at least return somekind of lazy object that can return the first byte. or validators for attribute data type should be aware of dynamic properties?

Reply:
@djay Every NamedBlobFile object has a data attribute and a getSize method. The base validator in zope.schema checks that value is an instance of the field's `_type`, so we don't need to check those explicitly.

We can't remove IFile (the data attribute) from INamedBlobFile; that's not a backwards compatible change. But you don't need to. I meant for you to change `schema = INamedBlobFile` to `schema = INamedTyped` on the field class, so that it's validating an interface that doesn't include `data` but without changing what's included in INamedBlobFile.

Reply:
that makes sense. I've made those changes